stateupgrade/state_upgrade.go

```diff
@@ -0,0 +1,45 @@
+// (c) 2023 Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package stateupgrade
```

Comment:
I agree that this would make more sense under a general upgrades package containing both state upgrades and precompiles. I'd like to avoid putting it directly under precompiles since it is not a precompile.

Reply:
what's our final decision with this?

Reply:
I'll defer to @darioush on this